igcollect/logfile_values.py

```diff
@@ -65,28 +65,66 @@ def get_timeshift(self):
                     return int(self.period[:-len(unit)].strip()) * mul
         return 0
 
+    def extract_value_from_field(self, field):
+        '''
+        Extract numeric value from field, handling key-value pairs and plain values
+        '''
+        # If it's a key-value pair, extract the value
+        if '=' in field:
+            key, value = field.split('=', 1)
+            # Remove quotes if present
+            value = value.strip('"')
+            try:
+                return float(value)
+            except ValueError:
+                return 0
+        else:
+            # Plain value
+            try:
+                return float(field)
+            except ValueError:
+                return 0
+
     def estimate_columns_value(self, fields):
         '''
         Apply some arithmetic on several columns if needed
-        Warning: Estimates regardless of arithmetics rules
+        Now handles key-value pairs properly
         '''
+        # Handle simple column reference first (most common case)
+        if self.column.isdigit():
+            try:
+                field = fields[int(self.column)]
+                return self.extract_value_from_field(field)
+            except (IndexError, ValueError):
+                return 0
+
+        # Handle complex expressions with arithmetic
         arr = [
             s for s in self.column if s.isdigit() or s in ['/', '*', '+', '-']
         ]
         result = 0
         for index, value in enumerate(arr):
             try:
                 if value.isdigit() and result == 0:
-                    result += float(fields[int(value)])
+                    field = fields[int(value)]
+                    result += self.extract_value_from_field(field)
                 elif value == '/':
-                    result = result / float(fields[int(arr[index + 1])])
+                    field = fields[int(arr[index + 1])]
+                    div_value = self.extract_value_from_field(field)
+                    if div_value != 0:
+                        result = result / div_value
+                    else:
+                        result = 0
                 elif value == '*':
-                    result = result * float(fields[int(arr[index + 1])])
+                    field = fields[int(arr[index + 1])]
+                    result = result * self.extract_value_from_field(field)
                 elif value == '+':
-                    result = result + float(fields[int(arr[index + 1])])
+                    field = fields[int(arr[index + 1])]
+                    result = result + self.extract_value_from_field(field)
                 elif value == '-':
-                    result = result - float(fields[int(arr[index + 1])])
-            except ZeroDivisionError:
+                    field = fields[int(arr[index + 1])]
+                    result = result - self.extract_value_from_field(field)
+            except (ZeroDivisionError, IndexError, ValueError):
```

Comment:
You are checking whatever the divisor is zero so you don't need to except ZeroDivisionError anymore.